CMakeLists.txt

```diff
@@ -31,10 +31,19 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
         compiler_not_supported("Your clang version ${CMAKE_CXX_COMPILER_VERSION} is too old.")
     endif()
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
-    if(NOT EXISTS "/usr/local/opt/llvm/bin/clang")
+    execute_process(
```

Comment:
Since I am a Cmake newbie and cannot judge it and the CI won't run this piece: have you tested it? :)

Reply:
I make a new build folder and tried to run cmake with Apple's clang. The error I got was:
```
  Apple's clang compiler is not supported because it is lacking support for
  recent C++ features.  Please install the official llvm/clang package using
  `install_dependencies.sh` or `brew install llvm --with-toolchain` and then
  run cmake with `-DCMAKE_C_COMPILER=/opt/homebrew/llvm/bin/clang
  -DCMAKE_CXX_COMPILER=/opt/homebrew/llvm/bin/clang++`.
```

Then, I changed `brew --prefix` to `unknown_command` which will fail to simulate brew was not installed. I got the following error:
```
  Apple's clang compiler is not supported because it is lacking support for
  recent C++ features.  Please install the official llvm/clang package using
  `install_dependencies.sh` or `brew install llvm --with-toolchain` and then
  run cmake with `-DCMAKE_C_COMPILER=<brew_library_path>/llvm/bin/clang
  -DCMAKE_CXX_COMPILER=<brew_library_path>/llvm/bin/clang++`.
```